public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -368,36 +368,34 @@ export class DashboardModel implements TimeModel {
       });
   }
 
-  private updateTemplatingSaveModelClone(
-    copy: any,
-    defaults: { saveTimerange: boolean; saveVariables: boolean } & CloneOptions
-  ) {
-    const originalVariables = this.originalTemplating;
+  private getTemplatingSaveModel(options: CloneOptions) {
+    const originalVariables = this.originalTemplating?.list ?? [];
     const currentVariables = this.getVariablesFromState(this.uid);
 
-    copy.templating = {
-      list: currentVariables.map((variable) =>
-        variableAdapters.get(variable.type).getSaveModel(variable, defaults.saveVariables)
-      ),
-    };
+    return {
+      list: currentVariables.map((variable) => {
+        const variableSaveModel = variableAdapters.get(variable.type).getSaveModel(variable, options.saveVariables);
 
-    if (!defaults.saveVariables) {
-      for (const current of copy.templating.list) {
-        const original = originalVariables.find(
-          ({ name, type }: any) => name === current.name && type === current.type
-        );
+        if (!options.saveVariables) {
+          const original = originalVariables.find(
+            ({ name, type }: any) => name === variable.name && type === variable.type
+          );
 
-        if (!original) {
-          continue;
-        }
+          if (!original) {
+            return variableSaveModel;
+          }
 
-        if (current.type === 'adhoc') {
-          current.filters = original.filters;
-        } else {
-          current.current = original.current;
+          if (variable.type === 'adhoc') {
+            variableSaveModel.filters = original.filters;
+          } else {
+            variableSaveModel.current = original.current;
+            variableSaveModel.options = original.options;
```

Comment:
Do we need to save the options? They are fetched anyways on load and https://github.com/grafana/grafana/pull/76403 should make sure the custom one is handled as well.

Reply:
aha, ok we need not to trigger the changes detection for every single dashboard after this is merged.